### PR TITLE
EVG-20892 Set users setting for spruce UI when clearing them

### PR DESCRIPTION
**Pull request by ZackarySantana**
EVG-20892

### Description
Recent returning interns have had their settings wiped when offboarded which caused them to use the old ui when they returned. This prevents that issue in the future by setting the spruce UI setting when offboarding a user.

### Testing
Unit tests

### CODE REVIEW

File: model/user/db.go [hunk 1]
```diff
@@ -488,18 +488,31 @@ func ClearLoginCache(user gimlet.User) error {
 
 // ClearUser clears one user's settings, roles, and login cache.
```
Comment (ablack12) [inline model/user/db.go:489]: we should update this to explain why we only clear some settings.

File: model/user/db.go [hunk 2]
```diff
@@ -488,18 +488,31 @@ func ClearLoginCache(user gimlet.User) error {
...
 		return errors.Wrap(err, "unsetting user settings")
 	}
-
-	return nil
+	setUpdate := bson.M{
+		"$set": bson.M{
+			SettingsKey: bson.M{
+				UseSpruceOptionsKey: bson.M{
+					SpruceV1Key: 1,
+				},
```
Comment (ablack12) [inline model/user/db.go:511]: 1 is only used for unset; for set, you want to use the actual value (true, false). In this case, I'm not actually sure waht you're going for. You can also just do this in one update: it would contain both the set and the unset (example: https://github.com/evergreen-ci/evergreen/blob/9f3350a1017ac4c63faa2480e273af4c7706a294/model/task/db.go#L176)

**Review by ablack12 — CHANGES_REQUESTED** [review]

File: model/user/db.go (hunk 1 above)
Comment (ZackarySantana) [reply model/user/db.go:489]: Let me know if I'm too wordy... Thought it better so no one has to investigate whyy this function does what it does.

File: model/user/db.go (hunk 2 above)
Comment (ablack12) [reply model/user/db.go:514]: Ah that makes sense, alas. What if we just didn't update this at all? Let them keep their default?

File: model/user/db.go [hunk 3]
```diff
@@ -487,19 +487,35 @@ func ClearLoginCache(user gimlet.User) error {
 }
 
 // ClearUser clears one user's settings, roles, and login cache.
```
Comment (ablack12) [inline model/user/db.go:489]: Delete this line

File: model/user/db.go [hunk 4]
```diff
@@ -487,19 +487,35 @@ func ClearLoginCache(user gimlet.User) error {
...
+// ClearUser clears the users sensitive settings (GitHub and Slack credentials),
+// roles, and invalidates their login cache. It sets their settings to use Spruce,
+// or else if this user ever becomes active, they will revert to the old UI.
 func ClearUser(userId string) error {
-	update := bson.M{
+	unsetUpdate := bson.M{
 		"$unset": bson.M{
-			SettingsKey:   1,
+			SettingsKey: bson.M{
```
Comment (ablack12) [inline model/user/db.go:496]: We should also clear notifications, timezone, and region.

File: model/user/db.go (hunk 2 above)
Comment (ablack12) [reply model/user/db.go:514]: Or if we're going to reset this anyway, we can just unset the entire user settings.

Comment (ablack12) [reply model/user/db.go:508]: You can just set this to true, not "true". SpruceV1 is a boolean.

File: model/user/db.go [hunk 5]
```diff
@@ -486,20 +486,30 @@ func ClearLoginCache(user gimlet.User) error {
...
+	setUpdate := bson.M{
+		"$set": bson.M{
+			SettingsKey: bson.M{
+				UseSpruceOptionsKey: bson.M{
+					SpruceV1Key: "true",
+				},
+			},
+		},
+	}
+	return errors.Wrap(UpdateOne(query, setUpdate), "unsetting user settings")
```
Comment (ablack12) [inline model/user/db.go:512]: this error message should be like "defaulting spruce setting" or something

**Review by ablack12 — APPROVED** [review]
LGTM with nits
